### ISSUE TITLE
Discussion: Where the data should come from?

### ISSUE THREAD

**Issue by gabriel19913** [okfn-brasil/perfil-politico#144]
I think a valid discussion we could have is where the data should come from.
Currently some data comes from brasil.io website and they are not updated frequently. So if the idea behind this project is to have something that is updated with certain frequency maybe would be interesting to have a scrip to get the data from TSE website and make it available in the cloud.

**Comment by cuducos** [comment]
If we can do that, why not just contribute to Brasil.IO which is open source, open data and hosting tons of datasets already? I think making them better is a win-win : )

**Comment by gabriel19913** [comment]
Hi @cuducos that would be nice too! I just don't know if this projetct should walk in parallel with brasil.io or if they should be independent from each other.
I think the community could contribute getting the data for brasil.io, but maybe the other contributors think is better to have the data in another place.
I really don't what is the best choice.

**Comment by cuducos** [comment]
>  I just don't know if this projetct should walk in parallel with brasil.io or if they should be independent from each other.

My opinion on that is that _preferably_ they should walk in parallel. Open-source in general (at least on the level we are in Open Knowledge Brasil and in Brasil.IO) usually suffer from a lack of support and hands-on to keep things up and running. Thus, bringing data collections, cleaning, and treatment to the scope of this project mean two things:

* increasing its complexity (or, in other words, risking decreasing it maintainability)
* capturing some volunteer (or even paid) work hours that would considerably help another open-source project (namely, Brasil.IO)

I think both consequences are not what Open Knowledge and Brasil.IO aspire for.

Surely I hear you: depending on an external project might make things move slower here — and I do agree with that. But I wouldn't assume the two consequences below before hearing from @turicas, for example, that Brasil.IO, for whatever reason, cannot accommodate contributions in a timely manner.

**TLDR** Getting back to the opening statement, __data comes from brasil.io website and they are not updated frequently__, I believe we could use this stamina to help them update the data we need more often instead of increasing the complexity of this project : )

**Comment by gabriel19913** [comment]
I got it @cuducos, thanks for your considerations. 
Thinking now maybe I was little bit harsh with this statement.
It wasn't my intention, I know how difficult is to work with open data specially when it depends on the goodwill of the public organizations. Besides that I imagine how hard is to get data for brasil.io specially now with the efforts aimed at covid's data.
I think you made really good points and we can help to update the data that is necessary for this project.

**Comment by hilam** [comment]
@cuducos 
> If we can do that, why not just contribute to Brasil.IO which is open source, open data and hosting tons of datasets already? I think making them better is a win-win : )

Because I agree with you, I tried, really, but the onboarding process is very difficult. 

In several failed attempts, during this week, I fell into a hell of unresolved dependencies. And the issue has six days without any response. https://github.com/turicas/brasil.io/issues/388

**Comment by cuducos** [comment]
> https://github.com/turicas/brasil.io/issues/388

Just a side note on Brasil.IO: this is not the repository I was suggesting to work on. This is the source code of their website, a Django instance with the dashboard, API and so on. Each dataset has its own repository with the code to collect and clean data.

If we go, for example, to their [_Eleições Brasil_ datasets page](https://brasil.io/dataset/eleicoes-brasil/candidatos/), there is the [URL for the repository that collects and cleans these datasets: `github.com/turicas/eleicoes-brasil`](https://github.com/turicas/eleicoes-brasil).

**Comment by hilam** [comment]
Excellent and necessary guidance, thank you.
